src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -35,6 +35,8 @@ public SearchParameterDefinitionManager(IModelInfoProvider modelInfoProvider)
 
         internal IDictionary<Uri, SearchParameterInfo> UrlLookup { get; set; }
 
+        internal IDictionary<string, IDictionary<string, SearchParameterInfo>> TypeLookup { get => _typeLookup; }
```

Comment:
Should we be returning a copy of the dictionary here so that we can prevent the caller from accidentally updating `_typeLookup`?

Reply:
I want the test cases to be able to update _typeLookup so they can insert new search params.  It will be somewhat limited as it is internal, but yes it does make it possible for other classes to make changes we may not want.